左からデフォルト値を上書きする

```
> c "mm/default_params.exs"
[Example]
> Example.func("a", "b")
["a", 2, 3, "b"]
["a", 2, 3, "b"]
> Example.func("a", "b", "c")
["a", "b", 3, "c"]
["a", "b", 3, "c"]
> Example.func("a", "b", "c","d")
["a", "b", "c", "d"]
["a", "b", "c", "d"]
> Example.func("a")
** (UndefinedFunctionError) function Example.func/1 is undefined or private. Did you mean one of:

      * func/2
      * func/3
      * func/4

    Example.func("a")
```

`func(p1, p2 \\ 2, p3 \\ 3, p4)`が`func(p1, p2)`を満たしてしまうのでエラーが発生する

```
> c "mm/default_params_error.exs"

== Compilation error in file mm/default_params_error.exs ==
** (CompileError) mm/default_params_error.exs:6: def func/2 conflicts with defaults from func/4
    mm/default_params_error.exs:6: (module)
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
** (CompileError)  compile error
    (iex) lib/iex/helpers.ex:183: IEx.Helpers.c/2
```

